### PR TITLE
More prefixes

### PR DESCRIPTION
**Pull request by sobychacko**
Redis sink, Hdfs Dataset sink, Trigger source

### CODE REVIEW

**Comment by markpollack** [comment]
Can we not call it just a 'redis' sink, this one is a redis-publisher'' sink... We should reserve the word 'redis' for the sink that writes to data structures.

**Comment by markfisher** [comment]
The 'redis' sink does more than PUBLISH. If a `key` is provided instead of a `topic` it _does_ write to a data structure. It can also write to a LIST if `queue` is provided (exactly one is required from topic, queue, or key)

**Comment by artembilan** [comment]
Deja Vu?
https://github.com/spring-cloud/spring-cloud-stream-app-starters/pull/138#discussion_r67230364

Doctor, ( @markpollack ?) I am being ignored. :smile: 

Although we should consider to add a `Map` functionality to that sink as well.

**Comment by markfisher** [comment]
+1 for adding support for writing to HASH data structures as well (either a fixed "field" or an expression for determining the "field" as well as the "value")

**Comment by markpollack** [comment]
It seems less complete the what was in XD no?

**Comment by markfisher** [comment]
XD's redis sink also has topic, queue, or key expressions

File: app-starters-common/app-starters-trigger-common/src/main/java/org/springframework/cloud/stream/app/trigger/SourcePayloadProperties.java [hunk 1]
```diff
@@ -22,7 +22,7 @@
 /**
  * @author Artem Bilan
  */
-@ConfigurationProperties
+@ConfigurationProperties("trigger.source")
```
Comment (ericbottard) [inline app-starters-common/app-starters-trigger-common/src/main/java/org/springframework/cloud/stream/app/trigger/SourcePayloadProperties.java:25]: why "source"?

Comment (sobychacko) [inline app-starters-common/app-starters-trigger-common/src/main/java/org/springframework/cloud/stream/app/trigger/SourcePayloadProperties.java:25]: there was already another @ConfigurationProperties with the trigger prefix and I didn't want to have a conflict with that.

**Comment by ericbottard** [comment]
Rebased and merged as cf2239b

**Pull request closed without merging**